NuRadioReco/modules/RNO_G/stationLDAVariables.py

```diff
@@ -0,0 +1,229 @@
+                shift = np.argmax(cor) - (len(ch.get_trace()) - 1)  # Calculate shift based on maximum correlation
+                
+                # Pad and shift the waveform to align with the reference sum
+                padded_wf = np.pad(ch.get_trace(), self.__pad_length, mode='constant')
+                aligned_wf = np.roll(padded_wf, shift)
+                
+                # Plot the current coherent sum before adding the next waveform
+                ax.plot(current_sum[self.__pad_length:-self.__pad_length], label='Current Coherent Sum', linestyle='--')
+                ax.plot(aligned_wf[self.__pad_length:-self.__pad_length], label=f'Next wf[{ch.get_id()}]', alpha=0.7)
+                
+                # Add the aligned waveform to the current sum
+                current_sum += aligned_wf
+                
+                # Customize plot
+                ax.legend(loc='upper right')
+                ax.set_xlabel('Sample')
+                ax.set_ylabel('Amplitude')
+                ax.set_title(f'Step {step + 1}: Adding wf[{ch.get_id()}]')
+            
+            # Save each coherent summing figure for the current reference
+            plt.tight_layout(rect=[0, 0, 1, 0.96])
+            # plt.savefig(f'coherent_sum_steps_reference_{ref_ch.get_id()}.png')
+            plt.show()
+
+    def coherent_sum(self, event, station, ref_ch):
+
+        self.sum_chan = ref_ch.get_trace()
+        channels = [ch for ch in station.iter_channels(use_channels=self.__channel_ids) if ch.get_id() != ref_ch.get_id()]
+                
+        for idx, ch in enumerate(channels):
+            cor = signal.correlate(np.abs(hilbert(self.sum_chan)), np.abs(hilbert(ch.get_trace())), mode="full")
+            lag = int(np.argmax((cor)) - (np.size(cor)/2.))
+            
+            aligned_wf = np.roll(ch.get_trace(), lag)
+            self.sum_chan += aligned_wf
+            
+    def impulsive_value(self, volts):
+
+        analytical_signal = hilbert(
+            volts
+        )  # compute analytic signal using hilbert transform from signal voltages
+        envelope = np.abs(analytical_signal)
+        maxv = np.argmax(envelope)
+        self.maxspot = (
+            maxv  ## index where the max voltage of the coherent sum is located
+        )
+        power_indexes = np.linspace(
+            0, len(envelope) - 1, len(envelope)
+        )  ## just a list of indices the same length as the array
+        closeness = list(
+            np.abs(power_indexes - maxv)
+        )  ## create an array containing index distance to max voltage (lower the value, the closer it is)
+
+        sorted_power = [x for _, x in sorted(zip(closeness, envelope))]
+        CDF = np.cumsum(sorted_power)
+        CDF = CDF / CDF[-1]
+
+        CDF_avg = (np.mean(np.asarray([CDF])) * 2.0) - 1.0
+        
+        self.CDF_avg = CDF_avg
+
+        if CDF_avg < 0:
+            CDF_avg = 0.0
+
+        return CDF_avg
```

Comment:
please do not use capital letters in variable names (in particular not as first letter). I would just write them all small.